### ISSUE TITLE
Assertions.fail() without a message throws PrecondtionViolationException

### ISSUE THREAD

**Issue by sbke-mms** [junit-team/junit5#3233]
ReportEntry requires the message not to be blank [(see ReportEntry.add())](https://github.com/junit-team/junit5/blob/1c294651ab3965a8bfe8909f6efdc56c5a47c4f0/junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java#L77) but [Assertions.fail()](https://github.com/junit-team/junit5/blob/1c294651ab3965a8bfe8909f6efdc56c5a47c4f0/junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java#L106-L118) without a message will set an empty String as value for the message which results in a PreconditionViolationException. This confused me a bit (I used it to test something out).

Maybe it's possible to add a default message like "Assertions.fail() called" when calling the plain `Assertions.fail()`. It's only a minor thing though since no one is probably using this in any productive way. I could add a PR for this issue if it's wanted or you can just close it if it is something that should stay this way.

**Comment by PhilipIngham** [comment]
Yeah that's no worries :) Just wanted to contribute something since I use JUnit a lot, and now I can say I have (or at least attempted, even if nothing moves forwards past discussions :) )
